app/scripts/lib/ppom/ppom-middleware.ts

```diff
@@ -68,11 +66,12 @@ export function createPPOMMiddleware<
         preferencesController.store.getState()?.securityAlertsEnabled;
 
       const { chainId } = networkController.state.providerConfig;
+      const isSupportedChain = await isChainSupported(chainId);
 
       if (
         !securityAlertsEnabled ||
         !CONFIRMATION_METHODS.includes(req.method) ||
-        !SECURITY_PROVIDER_SUPPORTED_CHAIN_IDS.includes(chainId)
+        !isSupportedChain
```

Comment:
Do we also need to update this check within `app/scripts/lib/transaction/util.ts` or we could move it inside `validateRequestWithPPOM`?

Reply:
I've updated the `validateSecurity, I think it is better to return early in this case. https://github.com/MetaMask/metamask-extension/pull/25716/commits/47cce7315573437388ce4c048f500957ada50944